src/mlpack/core/tree/vantage_point_tree/vantage_point_tree.hpp

```diff
@@ -0,0 +1,513 @@
+/**
+ * @file vantage_point_tree.hpp
+ *
+ * Definition of the vantage point tree.
+ */
+
+#ifndef MLPACK_CORE_TREE_VANTAGE_POINT_TREE_VANTAGE_POINT_TREE_HPP
+#define MLPACK_CORE_TREE_VANTAGE_POINT_TREE_VANTAGE_POINT_TREE_HPP
+
+#include "vantage_point_split.hpp"
+
+namespace mlpack {
+namespace tree /** Trees and tree-building procedures. */ {
+
+/**
+ * The vantage point tree is a variant of a binary space tree. The difference
+ * from BinarySpaceTree is a presence of points in intermediate nodes.
+ * If an intermediate node holds a point, this point is the centroid of the
+ * bound.
```

Comment:
We should probably reference Peter Yianilos's paper here (and possibly Jeffrey Uhlmann's too), and point out that the vantage point tree and the metric tree are the same thing.